backend/app/routes.py

```diff
@@ -159,16 +190,47 @@ def get_links_between_multi_nodes():
     node_ids = parse_get_links_between_multi_nodes_request_body(request.json)
     optimal_links_data_list = []
 
+    connection = getConnection()
+
     for i in range(len(node_ids) - 1):

```

Comment:
So this just iterates over consecutive nodes and does the routing separately for each pair. That's fine. But why duplicate the code from the above function? Let's reuse it!